### PR TITLE
Metadata improvements and poverty gap additions

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
* Minor change.
* Impacted periods: all.
* Impacted areas:
  - ```docs```
  - ```tests```
  - ```variables\household\attributes.py```
* Details:
  - Added README.md explaining usage of ```.vscode``` as per #50 
  - Added poverty gap variables for BHC and AHC poverty from #55 
  - Renamed poverty variables for #56 
  - Added unit tests for both variables added
  - In order to slightly address #54 , add metadata so that heroku can access the pslmodels repository (current prototype for legislation explorer is [here](https://openfisca-uk-explorer.herokuapp.com))

These changes:

- Impact the OpenFisca-UK public API (for instance renaming or removing a variable)
- Add a new feature (for instance adding a variable)
- Change non-functional parts of this repository (for instance editing the README)

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: openfisca serve --country-package openfisca_uk --port $PORT --bind 0.0.0.0:$PORT
```
Comment (MaxGhenis) [inline Procfile:1]: newline

File: openfisca_uk/variables/household/attributes.py [hunk 2]
```diff
@@ -181,5 +181,41 @@ def formula(household, period, parameters):
...
+
+
+class poverty_gap_bhc(Variable):
+    value_type = float
+    entity = Household
+    label = u"Positive financial gap between net household income and the poverty line"
+    definition_period = WEEK
+
+    def formula(household, period, parameters):
+        poverty_line = parameters(
```
Comment (MaxGhenis) [inline openfisca_uk/variables/household/attributes.py:195]: could an extra formula improve modularity for each `poverty_gap` formula? e.g.:

```
def poverty_gap(threshold, equivalization, net_income):
    poverty_line = threshold / equivalization
    return max_(0, poverty_line - net_income)
```

File: openfisca_uk/variables/person/taxes/income_tax.py [hunk 3]
```diff
@@ -315,7 +317,7 @@ def formula(person, period, parameters):
             "pension_contributions",
         ]
         total_deductions = add(person, period, DEDUCTIBLE)
-        return total_deductions
+        return max_(0, total_deductions)
```
Comment (MaxGhenis) [inline openfisca_uk/variables/person/taxes/income_tax.py:320]: interesting, do these deductions increase MTR heterogeneity?

File: tests/attributes/poverty_gap_ahc.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        age: 15
+    households:
+      household:
+        adults: [parent, second_adult]
+        children: child
+        household_net_income_ahc: 450 * 52
+  output:
+    household_equivalisation_bhc: 1.33
+    poverty_gap_ahc: 
+      week:2020-01-06: 0
```
Comment (MaxGhenis) [inline tests/attributes/poverty_gap_ahc.yaml:38]: newline

File: tests/attributes/poverty_gap_bhc.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        age: 15
+    households:
+      household:
+        adults: parent
+        children: child
+        household_net_income: 450 * 52
+  output:
+    household_equivalisation_bhc: 1.33
+    poverty_gap_bhc: 
+      week:2020-01-06: 0
```
Comment (MaxGhenis) [inline tests/attributes/poverty_gap_bhc.yaml:38]: newline

**Review by MaxGhenis — COMMENTED** [review]
nice updates nikhil

File: openfisca_uk/variables/household/attributes.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_uk/variables/household/attributes.py:195]: Yeah I like that, though I think it's best to keep all formulas OpenFisca standard - we could add a poverty_line formula for households, and have the poverty gap formula just use that

File: openfisca_uk/variables/person/taxes/income_tax.py (hunk 3 above)
Comment (nikhilwoodruff) [reply openfisca_uk/variables/person/taxes/income_tax.py:320]: Yeah I think they do a lot - maybe we should add a variable for taxable income after deductions, and use that for the MTR plots?

Comment (nikhilwoodruff) [reply openfisca_uk/variables/person/taxes/income_tax.py:320]: Actually, that probably erodes the meaning a bit though...

**Pull request merged**